api.go

```diff
@@ -215,25 +215,40 @@ func (h *Headscale) PollNetMapHandler(c *gin.Context) {
 		return
 	}
 
-	log.Printf("[%s] sending initial map", m.Name)
-	pollData <- *data
-
 	// We update our peers if the client is not sending ReadOnly in the MapRequest
 	// so we don't distribute its initial request (it comes with
 	// empty endpoints to peers)
-	if !req.ReadOnly {
-		peers, _ := h.getPeers(m)
-		h.pollMu.Lock()
-		for _, p := range *peers {
-			log.Printf("[%s] notifying peer %s (%s)", m.Name, p.Name, p.Addresses[0])
-			if pUp, ok := h.clientsPolling[uint64(p.ID)]; ok {
-				pUp <- []byte{}
-			} else {
-				log.Printf("[%s] Peer %s does not appear to be polling", m.Name, p.Name)
-			}
+
+	// Details on the protocol can be found in https://github.com/tailscale/tailscale/blob/main/tailcfg/tailcfg.go#L696
+	log.Printf("[%s] ReadOnly=%t   OmitPeers=%t    Stream=%t", m.Name, req.ReadOnly, req.OmitPeers, req.Stream)
+
+	if req.ReadOnly {
+		log.Printf("[%s] Client is starting up. Asking for DERP map", m.Name)
+		c.Data(200, "application/json; charset=utf-8", *data)
+		return
+	}
+	if req.OmitPeers {
```

Comment:
I *think* that the tailscale client currently only uses the combination of OmitPeers == true and Stream == false, and logically that seems to be the intention, but we should probably be explicit about that expectation How about we change this if to 

  if req.OmitPeers && ! req.Stream {
  ...
  } else if req.OmitPeers && req.Stream {
    log.Printf("[%s] Warning, ignoring request, don't know how to handle it", m.Name)
    // send 400 error code back with empty json response

    return
  }

Reply:
Implemented!